crates/conjure_core/src/rules/partial_eval.rs

```diff
@@ -40,6 +40,29 @@ fn partial_evaluator(expr: &Expr, _: &Model) -> ApplicationResult {
             }
         }
 
+        Product(m, vec) => {
```

Comment:
I don't think this handles the case where we have a 0 in the constants?

Also to note: we cannot just replace this with 0 since some of the expressions may be undefined. Only if they are all `can_be_undefined=false` we can.